.checkstyle.xml

```diff
@@ -0,0 +1,187 @@
+<?xml version="1.0"?>
+<!-- https://checkstyle.org -->
+<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
+<module name="Checker">
+  <!-- Checker properties                                        -->
+  <!-- See https://checkstyle.org/config.html#Checker_Properties -->
+  <property name="charset" value="UTF-8"/>
+  <property name="fileExtensions" value="java"/>
+  <property name="localeLanguage" value="en"/>
+  <property name="localeCountry" value="US"/>
+  <property name="severity" value="error"/>
+  <property name="tabWidth" value="4"/>
+
+  <!-- Checks whether files end with a new line                       -->
+  <!-- See https://checkstyle.org/config_misc.html#NewlineAtEndOfFile -->
+  <module name="NewlineAtEndOfFile"/>
+
+  <!-- Checks for size violations                   -->
+  <!-- See https://checkstyle.org/config_sizes.html -->
+  <module name="FileLength"/>
+  <module name="LineLength">
+    <property name="fileExtensions" value="java"/>
+    <property name="max" value="120"/>
+  </module>
+
+  <!-- Checks for whitespace                             -->
+  <!-- See https://checkstyle.org/config_whitespace.html -->
+  <module name="FileTabCharacter"/>
+  <module name="RegexpMultiline">
+    <property name="format" value="\n[\t ]*\r?\n[\t ]*\r?\n"/>
+    <property name="message" value="Multiple adjacent blank lines."/>
+  </module>
+  <module name="RegexpSingleline">
+    <property name="format" value="\s+$"/>
+    <property name="message" value="Line has trailing whitespace."/>
+  </module>
+
+  <!-- Checks for headers                            -->
+  <!-- See https://checkstyle.org/config_header.html -->
+  <!-- <module name="Header">
+    <property name="fileExtensions" value="java"/>
+    <property name="headerFile" value="LICENCE.txt"/>
+  </module> -->
+
+  <!-- Checks that a package-info.java file exists for each package  -->
+  <!-- See https://checkstyle.org/config_javadoc.html#JavadocPackage -->
+  <!-- <module name="JavadocPackage"/> -->
+
+  <module name="TreeWalker">
+
+    <!-- Checks for imports                             -->
+    <!-- See https://checkstyle.org/config_imports.html -->
+    <module name="AvoidStarImport"/>
+    <module name="IllegalImport"/>
+    <module name="ImportOrder">
+      <property name="groups" value="java,javax,*,net.minecraft,net.fabricmc"/>
```

Comment:
First party packages should probably have their own group as well, but hard to demo in an example project.